### DISCUSSION TITLE
Problem of select count after group

### DISCUSSION THREAD

**Discussion by myyrakle** [diesel-rs/diesel#2689] [Q&A] [answered]
The DB used is postgres, and the cargo.toml setting is 
diesel = {version= "1", features = ["postgres"] }

The table schema used for testing is as follows:
`table! {
    test (id) {
        id -> Int8,
        text -> Nullable<Text>,
        dead_yn -> Bool,
    }
}
`

And I tried to count after group by as follows.

```
    use crate::schema::test;
    use diesel::dsl::count;

    // select sum(id) from test
    let query = test::dsl::test
        .group_by(test::dsl::dead_yn)
        .select((test::dsl::dead_yn, count(test::dsl::id)));
    println!(
        "query {:?}",
        diesel::debug_query::<diesel::pg::Pg, _>(&query)
    );

    let result = query.get_results::<(bool, i64)>(connection).unwrap();
    println!("value {:?}", result);
```

A compilation error occurs.

> `error[E0277]: the trait bound `diesel::expression::count::count::count<diesel::sql_types::BigInt, schema::test::columns::id>: diesel::Expression` is not satisfied
>   --> src\main.rs:63:17
>    |
> 63 |         .select(&(test::dsl::dead_yn, count(test::dsl::id)));
>    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `NonAggregate` is not implemented for `diesel::expression::count::count::count<diesel::sql_types::BigInt, schema::test::columns::id>`
>    |
>    = note: required because of the requirements on the impl of `diesel::Expression` for `(schema::test::columns::dead_yn, diesel::expression::count::count::count<diesel::sql_types::BigInt, schema::test::columns::id>)`
>    = note: required because of the requirements on the impl of `diesel::Expression` for `&(schema::test::columns::dead_yn, diesel::expression::count::count::count<diesel::sql_types::BigInt, schema::test::columns::id>)`
> 
> error[E0277]: the trait bound `diesel::expression::count::count::count<diesel::sql_types::BigInt, schema::test::columns::id>: NonAggregate` is not satisfied
>   --> src\main.rs:69:24
>    |
> 69 |     let result = query.get_results::<(bool, i64)>(connection).unwrap();
>    |                        ^^^^^^^^^^^ the trait `NonAggregate` is not implemented for `diesel::expression::count::count::count<diesel::sql_types::BigInt, schema::test::columns::id>` 
>    |
>    = note: required because of the requirements on the impl of `diesel::Expression` for `(schema::test::columns::dead_yn, diesel::expression::count::count::count<diesel::sql_types::BigInt, schema::test::columns::id>)`
>    = note: required because of the requirements on the impl of `LoadQuery<_, (bool, i64)>` for `diesel::query_builder::SelectStatement<schema::test::table, query_builder::select_clause::SelectClause<&(schema::test::columns::dead_yn, diesel::expression::count::count::count<diesel::sql_types::BigInt, schema::test::columns::id>)>, query_builder::distinct_clause::NoDistinctClause, query_builder::where_clause::NoWhereClause, query_builder::order_clause::NoOrderClause, query_builder::limit_clause::NoLimitClause, query_builder::offset_clause::NoOffsetClause, query_builder::group_by_clause::GroupByClause<schema::test::columns::dead_yn>>`
> 
> error: aborting due to 2 previous errors; 2 warnings emitted
> 
> For more information about this error, try `rustc --explain E0277`.
> error: could not compile `rustywiki`
> 
> To learn more, run the command again with --verbose.
>

What am I doing wrong? 
Please help.

**Comment by weiznich** [comment, marked as answer]
`.group_by` is not part of the public API of any release yet. As consequence we do not support select clauses with mixed aggregate expressions yet. That all means that it is not possible to write such queries using the built-in DSL yet.

Our next feature release will add support for this offically.

**Reply by myyrakle** [reply]
Thanks for the kind answer.
We are looking forward to the next version.
